### PR TITLE
RXO-11159: Add SAC tests for no access

### PR DESCRIPTION
**Pull request by janisz**
## Description

Adjust tests

## Testing Performed

CI

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2023** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2023" target="_blank">Graphite</a> 
    * **PR #2024** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2024" target="_blank">Graphite</a> 
      * **PR #2025** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2025" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2025?utm_source=stack-comment).

**Review by rhybrillou — COMMENTED** [review]
Can you please ensure the test for `Delete` using `withNoAccessCtx` does test the complete behaviour of the `Delete` function (should leave any existing item untouched) on top of the error return behaviour of the function ?

File: tools/generate-helpers/pg-table-bindings/store_test.go.tpl [hunk 1]
```diff
@@ -129,9 +129,11 @@ func (s *{{$namePrefix}}StoreSuite) TestStore() {
 	s.False(exists)
 	s.Nil(found{{.TrimmedType|upperCamelCase}})
 
-    {{- if or (.Obj.IsGloballyScoped) (.Obj.HasPermissionChecker) }}
+	{{- if or (.Obj.IsGloballyScoped) (.Obj.HasPermissionChecker) }}
     s.ErrorIs(store.Delete(withNoAccessCtx, {{template "paramList" $}}), sac.ErrResourceAccessDenied)
-    {{- end }}
+	{{- else }}
+	s.NoError(store.Delete(withNoAccessCtx, {{template "paramList" $}}))
+	{{- end }}
```
[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store_test.go.tpl:136]: This test works as you described. below is the code that check if "deleted" object still exists.

**Comment by janisz** [comment]
Looks like some test are failing but in no pg world so failures are unrelated to this PR

**Review by rhybrillou — APPROVED** [review]

**Pull request merged**